assets/js/payment-method-extensions/payment-methods/stripe/apple-pay/apple-pay-express.js

```diff
@@ -223,17 +221,33 @@ const ApplePayExpressComponent = ( {
 		}
 	};
 
+	const onPaymentProcessing = () => {
+		const handlers = eventHandlers.current;
+		if (
+			typeof handlers.sourceEvent === 'function' &&
+			applePayProcessing
+		) {
+			return {
+				billingData: getBillingData( handlers.sourceEvent ),
+				paymentMethodData: getPaymentMethodData(
+					handlers.sourceEvent,
+					PAYMENT_METHOD_NAME
+				),
+			};
+		}
+		return true;
+	};
+
 	const onCheckoutComplete = ( forSuccess = true ) => () => {
 		const handlers = eventHandlers.current;
 		if (
 			typeof handlers.sourceEvent === 'function' &&
-			currentPaymentStatus.current.currentStatus.isSuccessful
+			applePayProcessing
 		) {
 			if ( forSuccess ) {
 				completePayment( handlers.sourceEvent );
-			} else {
-				abortPayment( handlers.sourceEvent );
 			}
+			abortPayment( handlers.sourceEvent );
```

Comment:
Why is `abortPayment` outside the `else` clause, now? Shouldn't it be called only when `forSuccess` is false?

Reply:
My guess was to reuse `setApplePayProcessing` but the side effect is that payment would complete, then fail. So looks like a bug.

Reply:
yeah the function should return or exist at `completePayment( handlers.sourceEvent );` so that this doesn't get called.

Reply:
yup a bug, I noticed and fixed in another in progress pull, but I'll add the fix here too.

Reply:
See e04b413c6 for fix. Also noticed that the completePayment call wasn't setting `applePayProcessing` to false.